rest/oidc_api_test.go

```diff
@@ -1,139 +1,612 @@
+	require.NoError(t, err, "Error mocking fake authorization server")
+	defer mockAuthServer.Close()
+
+	options := NewMockOptions()
+	defer options.Clear()
+	options["issuerGoogle"] = mockAuthServer.URL + "/google"
+	options["issuerFacebook"] = mockAuthServer.URL + "/facebook"
+	issuerGoogle := options["issuerGoogle"].(string)
+	issuerFacebook := options["issuerFacebook"].(string)
 
 	// Default OpenID Connect Provider
 	providerGoogle := auth.OIDCProvider{
-		Name:          providerNameGoogle,
+		Name:          "google",
 		Issuer:        issuerGoogle,
-		ClientID:      &clientIDGoogle,
-		ValidationKey: &validationKeyGoogle,
-		DiscoveryURI:  issuerGoogle + wellKnownPath,
+		ClientID:      "foo",
+		ValidationKey: base.StringPointer("bar"),
+		DiscoveryURI:  issuerGoogle + auth.DiscoveryConfigPath,
 	}
 
 	// Non-default OpenID Connect Provider
-	providerSalesforce := auth.OIDCProvider{
-		Name:          providerNameSalesforce,
-		Issuer:        issuerSalesforce,
-		ClientID:      &clientIDSalesforce,
-		ValidationKey: &validationKeySalesforce,
-		DiscoveryURI:  issuerSalesforce + wellKnownPath,
+	providerFacebook := auth.OIDCProvider{
+		Name:          "facebook",
+		Issuer:        issuerFacebook,
+		ClientID:      "foo",
+		ValidationKey: base.StringPointer("bar"),
+		DiscoveryURI:  issuerFacebook + auth.DiscoveryConfigPath,
 	}
 
-	providerMap := auth.OIDCProviderMap{
-		providerNameGoogle:     &providerGoogle,
-		providerNameSalesforce: &providerSalesforce,
-	}
-
-	openIDConnectOptions := auth.OIDCOptions{Providers: providerMap, DefaultProvider: &providerGoogle.Name}
+	providers := auth.OIDCProviderMap{"google": &providerGoogle, "facebook": &providerFacebook}
+	openIDConnectOptions := auth.OIDCOptions{Providers: providers, DefaultProvider: &providerGoogle.Name}
 	rtConfig := RestTesterConfig{DatabaseConfig: &DbConfig{OIDCConfig: &openIDConnectOptions}}
 	rt := NewRestTester(t, &rtConfig)
 	defer rt.Close()
 
-	t.Run("default provider configured but current provider is not default", func(t *testing.T) {
-		// When multiple providers are defined, default provider is specified and the current provider is
-		// not default, then current provider should be added to the generated OpenID Connect callback URL.
-		resp := rt.SendAdminRequest(http.MethodGet, "/db/_oidc?provider=salesforce&offline=true", "")
-		require.Equal(t, http.StatusFound, resp.Code)
-		location := resp.Header().Get(headerLocation)
-		require.NotEmpty(t, location, "Location should be available in response header")
-		locationURL, err := url.Parse(location)
-		require.NoError(t, err, "Location header should be a valid URL")
-		redirectURI := locationURL.Query().Get(authRedirectURI)
-		require.NotEmpty(t, location, "redirect_uri should be available in auth URL")
-		redirectURL, err := url.Parse(redirectURI)
-		require.NoError(t, err, "redirect_uri should be a valid URL")
-		assert.Equal(t, providerSalesforce.Name, redirectURL.Query().Get(authProvider))
-	})
+	type test struct {
```

Comment:
'wantProviderName' and 'expectedProvider' appear to be redundant. 

inputRequestURL and 'oidcChallenge' are also redundant.  Cleaning these up will make the tests easier to understand.